test/test_toroid.py

```diff
@@ -0,0 +1,183 @@
+# A script to run test cases of toroid-ray.py
+
+
+# TODO: do this in a more pythonic way
+import os
+import sys
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
+parent_dir = os.path.join(current_dir, "..")
+sys.path.insert(0, parent_dir)
+#
+
+import math
```

Comment:
Do you need al these imports? Can you just import a single thing out of some of them rather than the whole module?